### PR TITLE
tech debt: make cleanup

### PR DESCRIPTION
**Pull request by annastopel**
- remove stale make targets
       - add make target for cleanup docker cache on clsuter-down cluster-down-purge
       - minor fixes
       - update hack/build/README.md

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Content:
       - remove stale make targets
       - add make target for cleanup docker cache on clsuter-down cluster-down-purge
       - minor fixes
       - update hack/build/README.md

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by annastopel** [comment]
ci test please

**Comment by igoihman** [comment]
@annastopel These are great changes! Can you split it to a few commits? one for every change? It will make the review much more easy.

File: hack/build/cleanup_docker.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+REPO=
```
Comment (awels) [inline hack/build/cleanup_docker.sh]: You can replace the whole setTag function with this:

REPO=${DOCKER_REPO:-"localhost"}

That basically means REPO = ${DOCKER_REPO} if not found, use "localhost"

File: hack/build/cleanup_docker.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+function setTag {
+  if [ "$1" == "" ]; then
+      REPO="localhost"
+  else
+      REPO=$1
+  fi
+}
+
+function dockerCleanup {
```
Comment (awels) [inline hack/build/cleanup_docker.sh]: What does this do that "docker system prune -a" doesn't do?

**Comment by annastopel** [comment]
Prune will remove everything including k8s images, which can be reused in
opposite to cdi images that are built every cluster-sync and are set in
repo associated with generated local registry port of this specific cluster
instance

File: hack/build/cleanup_docker.sh (hunk 2 above)
Comment (annastopel) [reply hack/build/cleanup_docker.sh]: > What does this do that "docker system prune -a" doesn't do?

Prune will remove everything including k8s images, which can be reused in
opposite to cdi images that are built every cluster-sync and are set in
repo associated with generated local registry port of this specific cluster
instance

File: hack/build/cleanup_docker.sh (hunk 1 above)
Comment (annastopel) [reply hack/build/cleanup_docker.sh]: > You can replace the whole setTag function with this:
> 
> REPO=${DOCKER_REPO:-"localhost"}
> 
> That basically means REPO = ${DOCKER_REPO} if not found, use "localhost"

Nice! Updated

File: hack/build/cleanup_docker.sh (hunk 2 above)
Comment (mhenriks) [reply hack/build/cleanup_docker.sh]: this is a very cool feature.  But I think it should only prune images for the current (soon to be exiting) instance of the of the docker registry.  Can call gocli to get the registry port.  See here:  https://github.com/kubevirt/containerized-data-importer/blob/master/cluster/sync.sh#L16-L17

Comment (annastopel) [reply hack/build/cleanup_docker.sh]: > this is a very cool feature. But I think it should only prune images for the current (soon to be exiting) instance of the of the docker registry. Can call gocli to get the registry port. See here: https://github.com/kubevirt/containerized-data-importer/blob/master/cluster/sync.sh#L16-L17

Done. Please review

**Pull request merged**